# DNA-sequencing-classifier-using-NLP
DNA sequence data is most commonly represented in a format called as "fasta" format, which is a single line prefixed by ">" that contains annotations followed by another line that contains the actual sequence. A single file can contain one or more DNA sequences. 
Now, since the DNA sequence is just a combination of A-T-G-C, we have to convert them as usable features for the Machine Learning algorithms to understand and interpret. In order to do this, we have three commonly used approaches;  1) Use NLP methods and treat DNA sequence as a text language, also called as K-Mer counting, 2) Encoding the DNA sequence information as an ordinal vector which will convert this into an ordinal feature to be fed into ML algorithms, and 3) Using One-hot-encoding for A-T-G-C and converting it into array to be fed into deep learning algorithms.
It has to noted that excepting the NLP approach, the other two approaches does not produce equal length vectors. In order to get vectors of uniform length from the other two encoding appraches, truncating the sequences or padding needs to performed.
Here, we use the NLP approach to classify the DNA sequences.
